src/controllers/exercisesController.js

```diff
@@ -7,14 +7,18 @@ const Topic = require('../models/Topic');
 
 class ExercisesController {
   async postExerciseUser(exerciseId, userId) {
-    const exercise = Exercise.findByPk(exerciseId);
+    const exercise = await Exercise.findByPk(exerciseId);
     if (!exercise) throw new InexistingId();
-    await ExerciseUser.findOrCreate({
+    const [foundExerciseUser, createdExerciseUser] = await ExerciseUser.findOrCreate({
       where: {
         exerciseId,
         userId,
       },
     });
+
+    if (!createdExerciseUser) {
+      await foundExerciseUser.destroy();
+    }
```

Comment:
Aqui o createdExerciseUser for falso (ou seja, nao tinha nenhum exerciseUser) ele vai la e destrói o foundExerciseUser q foi criado, né? Mas se é isso não era mais facil fazer só um findOne?

Reply:
Com o findOne aí se não foi encontrado teria que criar. A complexidade das duas é a mesma :)